modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/resource/test/HeadlessBuilderResourceTest.java

```diff
@@ -1584,6 +1587,121 @@ public void testGetWithSystemObjectFields() throws Exception {
+			ObjectFieldValidationConstants.BUSINESS_TYPE_LONG_VALUE_MAX);
+		String longTextPropertyValue = RandomTestUtil.randomString();
+		Serializable multiselectPicklistPropertyValue =
+			(Serializable)TransformUtil.transform(
+				Arrays.asList(ListTypeValue.VALUE1, ListTypeValue.VALUE3),
+				ListTypeValue::name);
+		String picklistPropertyValue = ListTypeValue.VALUE1.name();
+		double precisionDecimalPropertyValue = RandomTestUtil.randomDouble();
+		String richTextPropertyValue = RandomTestUtil.randomString();
+		String textPropertyValue = RandomTestUtil.randomString();
+		String textUniquePropertyValue = "Unique field value";
+
+		JSONObject jsonObject = HTTPTestUtil.invokeToJSONObject(
+			JSONUtil.put(
+				"attachmentProperty", attachmentPropertyIdValue
+			).put(
+				"booleanProperty", booleanPropertyValue
+			).put(
+				"dateProperty", datePropertyValue
+			).put(
+				"dateTimeProperty", dateTimePropertyValue
+			).put(
+				"decimalProperty", decimalPropertyValue
+			).put(
+				"integerProperty", integerPropertyValue
+			).put(
+				"longIntegerProperty", longIntegerPropertyValue
+			).put(
+				"longTextProperty", longTextPropertyValue
+			).put(
+				"multiselectPicklistProperty", multiselectPicklistPropertyValue
+			).put(
+				"picklistProperty", picklistPropertyValue
+			).put(
+				"precisionDecimalProperty", precisionDecimalPropertyValue
+			).put(
+				"richTextProperty", richTextPropertyValue
+			).put(
+				"textProperty", textPropertyValue
+			).put(
+				"textUniqueProperty", textUniquePropertyValue
+			).toString(),
+			StringBundler.concat("c/", _BASE_URL_1, "/test"), Http.Method.POST);
+
+		JSONAssert.assertEquals(
+			JSONUtil.put(
+				"booleanProperty", booleanPropertyValue
+			).put(
+				"dateProperty", datePropertyValue
+			).put(
+				"dateTimeProperty", dateTimePropertyValue
+			).put(
+				"integerProperty", integerPropertyValue
+			).put(
+				"longTextProperty", longTextPropertyValue
+			).put(
+				"multiselectPicklistProperty",
+				JSONUtil.putAll(
+					JSONUtil.put(
+						"key", ListTypeValue.VALUE1
+					).put(
+						"name", ListTypeValue.VALUE1
+					),
+					JSONUtil.put(
+						"key", ListTypeValue.VALUE3
+					).put(
+						"name", ListTypeValue.VALUE3
+					))
+			).put(
+				"picklistProperty",
+				JSONUtil.put(
+					"key", ListTypeValue.VALUE1
+				).put(
+					"name", ListTypeValue.VALUE1
+				)
+			).put(
+				"richTextProperty", richTextPropertyValue
+			).put(
+				"textProperty", textPropertyValue
+			).put(
+				"textUniqueProperty", textUniquePropertyValue
+			).toString(),
+			jsonObject.toString(), JSONCompareMode.LENIENT);
+
+		Assert.assertEquals(
+			decimalPropertyValue, jsonObject.getDouble("decimalProperty"),
+			0.0000000000000001);
+		Assert.assertEquals(
+			longIntegerPropertyValue,
+			jsonObject.getLong("longIntegerProperty"));
+		Assert.assertEquals(
+			precisionDecimalPropertyValue,
+			jsonObject.getDouble("precisionDecimalProperty"),
+			0.0000000000000001);
+	}
```

Comment:
Maybe we should validate that an Object Entry has been created (or invoking the GET endpoint to validate it has been stored somewhere)